src/app/qgsstatisticalsummarydockwidget.cpp

```diff
@@ -129,39 +113,36 @@ void QgsStatisticalSummaryDockWidget::refreshStatistics()
     return;
   }
 
-  // non numeric field?
-  bool isNumeric = true;
-  QVariant::Type fieldType = QVariant::Double;
+  // determine field type
+  mFieldType = QVariant::Double;
   if ( !mFieldExpressionWidget->isExpression() )
   {
     QString field = mFieldExpressionWidget->currentField();
-    fieldType = mLayer->fields().field( mLayer->fields().lookupField( field ) ).type();
-    if ( fieldType == QVariant::String || fieldType == QVariant::Date || fieldType == QVariant::DateTime )
-    {
-      isNumeric = false;
-    }
+    mFieldType = mLayer->fields().field( mLayer->fields().lookupField( field ) ).type();
   }
 
+  refreshStatisticsMenu();
+
   bool selectedOnly = mSelectedOnlyCheckBox->isChecked();
 
-  if ( isNumeric )
+  switch ( mFieldType )
   {
-    updateNumericStatistics( selectedOnly );
-  }
-  else
-  {
-    switch ( fieldType )
-    {
-      case QVariant::String:
-        updateStringStatistics( selectedOnly );
-        break;
-      case QVariant::Date:
-      case QVariant::DateTime:
-        updateDateTimeStatistics( selectedOnly );
-        break;
-      default:
-        break;
-    }
+    case QVariant::Int:
```

Comment:
Can we reuse QgsField::isNumeric() instead? We're always getting burnt by numeric field types, which is why QgsField::isNumeric was introduced.

Otherwise I'd suggest adding a private method here which maps QVariant::Type to a private QgsStatisticalSummaryDockWidget::DataType (Numeric/String/DateTime) to avoid the multiple places in this code where the field type is mapped to a corresponding set of display stats.